src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -538,30 +556,54 @@ int32_t NaClSysOpen(struct NaClAppThread  *natp,
   char                 path[NACL_CONFIG_PATH_MAX];
   nacl_host_stat_t     stbuf;
   int                  allowed_flags;
+  const char           glibc_prefix[] = "/lib/glibc/";
+  const char           tls_prefix[] = "/lib/glibc/tls/";
+  const size_t         tls_start_idx = strlen(glibc_prefix);
+  const size_t         tls_end_idx = strlen(tls_prefix);
+
+  /* this is the virtual fd returned to the cage */
+  int                  fd_retval;
 
-  NaClLog(3, "NaClSysOpen(0x%08"NACL_PRIxPTR", "
+  NaClLog(2, "NaClSysOpen(0x%08"NACL_PRIxPTR", "
           "0x%08"NACL_PRIxPTR", 0x%x, 0x%x)\n",
-          (uintptr_t) natp, (uintptr_t) pathname, flags, mode);
+          (uintptr_t)natp, (uintptr_t)pathname, flags, mode);
 
-  retval = CopyPathFromUser(nap, path, sizeof path, (uintptr_t) pathname);
-  if (0 != retval)
+  retval = CopyPathFromUser(nap, path, sizeof path, (uintptr_t)pathname);
+
+  /*
+   * TODO:
+   * find a cleaner method to prevent the
+   * runtime linker from searching for libc
+   * in /lib/glibc/tls/ (this removes the
+   * tls/ path component).
+   *
+   * -jp
+   */
+  if (!memcmp(path, tls_prefix, sizeof tls_prefix - 1)) {
```

Comment:
Possible multi-byte char encoding issue ahead?

Reply:
The glibc build always uses `/lib/glibc/tls` so it shouldn't be a problem.